### PR TITLE
FT: Add replication of object tags

### PR DESCRIPTION
**Pull request by bennettbuchanan**

### CODE REVIEW

**Comment by bennettbuchanan** [comment]
@ironman-machine 
SCALITY_ARSENAL_BRANCH=ft/S3C-972/add-datastoreVersionId-to-replicationInfo
try

**Comment by bennettbuchanan** [comment]
@ironman-machine try
SCALITY_ARSENAL_BRANCH=
try

**Comment by bennettbuchanan** [comment]
@ironman-machine try
SCALITY_ARSENAL_BRANCH=
try

File: lib/routes/routeBackbeat.js [hunk 1]
```diff
@@ -373,20 +380,78 @@ function completeMultipartUpload(request, response, log, callback) {
...
+    const err = _checkMultipleBackendRequest(request, log);
+    if (err) {
+        return callback(err);
+    }
+    const storageLocation = request.headers['x-scal-storage-class'];
+    const dataStoreVersionId = request.headers['x-scal-data-store-version-id'];
+    const data = [];
+    request.on('data', chunk => data.push(chunk));
+    request.on('end', () =>
+        parseTagXml(Buffer.concat(data), log, (err, tags) => {
```
Comment (rahulreddy) [inline lib/routes/routeBackbeat.js]: > If totalLength is not provided, it is calculated from the Buffer instances in list. This however causes an additional loop to be executed in order to calculate the totalLength, so it is faster to provide the length explicitly if it is already known.

You should aggregate each chunk's length and then provide it to the Buffer.concat method.

File: lib/api/objectPutTagging.js [hunk 2]
```diff
@@ -78,10 +78,11 @@ function objectPutTagging(authInfo, request, log, callback) {
             const params = objectMD.versionId ? { versionId:
               objectMD.versionId } : {};
             const replicationInfo =
-                getReplicationInfo(objectKey, bucket, true);
+                getReplicationInfo(objectKey, bucket, true, 0, 'PUT_TAGGING');
```
Comment (rahulreddy) [inline lib/api/objectPutTagging.js]: Please create a constant after the requires, this way we don't have to hunt for these constants to maintain them.
```js
const REPLICATION_ACTION = 'PUT_TAGGING';
```

File: lib/api/objectDeleteTagging.js [hunk 3]
```diff
@@ -71,10 +71,12 @@ function objectDeleteTagging(authInfo, request, log, callback) {
             objectMD.tags = {};
             const params = objectMD.versionId ? { versionId:
               objectMD.versionId } : {};
-            const replicationInfo = getReplicationInfo(objectKey, bucket, true);
+            const replicationInfo = getReplicationInfo(objectKey, bucket, true,
+                0, 'DELETE_TAGGING');
```
Comment (rahulreddy) [inline lib/api/objectDeleteTagging.js]: Please create a constant after the requires, this way we don't have to hunt for these constants to maintain them.
```js
const REPLICATION_ACTION = 'DELETE_TAGGING';
```

Comment (bennettbuchanan) [reply lib/api/objectDeleteTagging.js]: Updated.

File: lib/api/objectPutTagging.js (hunk 2 above)
Comment (bennettbuchanan) [reply lib/api/objectPutTagging.js]: Updated.

File: lib/routes/routeBackbeat.js (hunk 1 above)
Comment (bennettbuchanan) [reply lib/routes/routeBackbeat.js]: Updated.

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Pull request merged**